### PR TITLE
chore: refactor checkFileSize function with async file reading 

### PR DESCRIPTION
**Pull request by kirklin**
Optimized the `checkFileSize` function in the `size-checks` script to improve its performance

### CODE REVIEW

File: scripts/check-size.js [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
+    ])
+    const gzippedSize = (gzipped.length / 1024).toFixed(2) + 'kb'
+    const compressedSize = (compressed.length / 1024).toFixed(2) + 'kb'
+    console.log(
+      `${chalk.gray(
+        chalk.bold(path.basename(filePath))
+      )} min:${minSize} / gzip:${gzippedSize} / brotli:${compressedSize}`
+    )
+  } catch (err) {
+    console.error(err)
```
Comment (posva) [inline scripts/check-size.js:23]: let's remove the try/catch. It should crash the program if it throws

**Review by posva — CHANGES_REQUESTED** [review]

File: scripts/check-size.js [hunk 2]
```diff
@@ -1,18 +1,14 @@
-const fs = require('fs')
+const fs = require('fs').promises
 const path = require('path')
 const chalk = require('chalk')
-const { gzipSync } = require('zlib')
+const { gzip } = require('zlib')
 const { compress } = require('brotli')
 
-function checkFileSize(filePath) {
-  if (!fs.existsSync(filePath)) {
```
Comment (posva) [inline scripts/check-size.js:8]: can you add back the removed check?

**Review by posva — CHANGES_REQUESTED** [review]

**Comment by posva** [comment]
I think you have a checkbox to let maintainer edits unchecked

**Comment by kirklin** [comment]
Oh, I accidentally closed it. Now I have reopened it.

**Pull request closed without merging**